### PR TITLE
Update citing information to include manual

### PR DESCRIPTION
**Pull request by JDBetteridge**
# Description
Update all citation information to include the Firedrake User Manual

## Associated Pull Requests:
- [PR#2784](https://github.com/firedrakeproject/firedrake/pull/2784) should go in first!

## Fixes Issues:
- N/A

# Checklist for author:

- [x] I have linted the codebase (`make lint` in the `firedrake` source directory).
- [x] My changes generate no new warnings.
- [ ] ~All of my functions and classes have appropriate docstrings.~
- [x] I have commented my code where its purpose may be unclear.
- [x] I have included and updated any relevant documentation.
- [x] Documentation builds locally (`make linkcheck; make html; make latexpdf` in `firedrake/docs` directory)
- [ ] ~I have added tests specific to the issues fixed in this PR.~
- [ ] ~I have added tests that exercise the new functionality I have introduced~
- [ ] ~Tests pass locally (`pytest tests` in the `firedrake` source directory) (useful, but not essential if you don't have suitable hardware).~
- [x] I have performed a self-review of my own code using the below guidelines.

# Checklist for reviewer:

- [ ] Docstrings present
- [ ] New tests present
- [ ] Code correctly commented
- [ ] No bad "code smells"
- [ ] No issues in parallel
- [ ] No CI issues (excessive parallelism/memory usage/time/warnings generated)
- [ ] Upstream/dependent branches and PRs are ready

Feel free to add reviewers if you know there is someone who is already aware of this work.

Please open this PR initially as a draft and mark as ready for review once CI tests are passing.

### CODE REVIEW

File: CITATION.rst [hunk 1]
```diff
@@ -0,0 +1,36 @@
+..
+   This file is generated by team2.py using the citations.rst_t template and team.ini.
+      DO NOT EDIT DIRECTLY
+   To add your name and/or institution edit the relevant sections of
+      docs/source/team.ini
```
Comment (dham) [inline CITATION.rst:5]: should autogenerated files be in the repo?

File: CITATION.rst [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+================
+Citing Firedrake
+================
+
+If you publish results using Firedrake, we would be grateful if you would cite the Firedrake user manual:
+
+  @manual{FiredrakeUserManual,
+    title		     = {Firedrake User Manual},
+    author		   = {David Ham and Paul Kelly and Lawrence Mitchell and Colin Cotter and Rob Kirby and Koki Sagiyama and Nacime Bouziani and Sophia Vorderwuelbecke and Tom Gregory and Jack Betteridge and Daniel Shapero and Reuben W. Nixon-Hill and Connor Ward and Patrick E. Farrell and Pablo Brubeck and India Marsden and Thomas Gibson and Miklós Homolya and Tianjiao (TJ) Sun and Andrew McRae and Fabio Luporini and Alastair Gregory and Michael Lange and Simon Funke and Florian Rathgeber and Doru Bercea and Graham Markall},
```
Comment (dham) [inline CITATION.rst:15]: Publication names in the bibtex entry please.

**Review by dham — DISMISSED** [review]

File: CITATION.rst (hunk 1 above)
Comment (JDBetteridge) [reply CITATION.rst:5]: In general no, but CITATION.rst and AUTHORS.rst exist to enable Github functionality and must be committed.

File: CITATION.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+If you publish results using Firedrake, we would be grateful if you would cite the Firedrake user manual::
+
+  @manual{FiredrakeUserManual,
+    title        = {Firedrake User Manual},
+    author       = {David A. Ham and Paul H. J. Kelly and Lawrence Mitchell and Colin J. Cotter and Rob C. Kirby and Koki Sagiyama and Nacime Bouziani and Sophia Vorderwuelbecke and Thomas J. Gregory and Jack Betteridge and Daniel R. Shapero and Reuben W. Nixon-Hill and Connor J. Ward and Patrick E. Farrell and Pablo D. Brubeck and India Marsden and Thomas H. Gibson and Miklós Homolya and Tianjiao Sun and Andrew T. T. McRae and Fabio Luporini and Alastair Gregory and Michael Lange and Simon W. Funke and Florian Rathgeber and Gheorghe-Teodor Bercea and Graham R. Markall},
+    organization = {Imperial College London and University of Oxford and Baylor University and University of Washington},
+    edition      = {First edition},
+    year         = {2023},
+    month        = {5},
+    doi          = {pending},
```
Comment (dham) [inline CITATION.rst:20]:
```suggestion
```

File: docs/source/citation.rst_t [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+If you publish results using Firedrake, we would be grateful if you would cite the Firedrake user manual::
+
+  @manual{FiredrakeUserManual,
+    title        = {Firedrake User Manual},
+    author       = {{ '{' }}{{ author }}{{ '}' }},
+    organization = {{ '{' }}{{ institution }}{{ '}' }},
+    edition      = {First edition},
+    year         = {2023},
+    month        = {5},
+    doi          = {pending},
```
Comment (dham) [inline docs/source/citation.rst_t:20]:
```suggestion
```

File: docs/source/citing.rst [hunk 5]
```diff
@@ -7,11 +7,21 @@
...
+would cite the Firedrake user manual::
+
+  @manual{FiredrakeUserManual,
+    title        = {Firedrake User Manual},
+    author       = {David A. Ham and Paul H. J. Kelly and Lawrence Mitchell and Colin J. Cotter and Rob C. Kirby and Koki Sagiyama and Nacime Bouziani and Sophia Vorderwuelbecke and Thomas J. Gregory and Jack Betteridge and Daniel R. Shapero and Reuben W. Nixon-Hill and Connor J. Ward and Patrick E. Farrell and Pablo D. Brubeck and India Marsden and Thomas H. Gibson and Miklós Homolya and Tianjiao Sun and Andrew T. T. McRae and Fabio Luporini and Alastair Gregory and Michael Lange and Simon W. Funke and Florian Rathgeber and Gheorghe-Teodor Bercea and Graham R. Markall},
+    organization = {Imperial College London and University of Oxford and Baylor University and University of Washington},
+    edition      = {First edition},
+    year         = {2023},
+    month        = {5},
+    doi          = {pending},
```
Comment (dham) [inline docs/source/citing.rst:19]:
```suggestion
```

**Review by dham — DISMISSED** [review]

File: docs/source/_static/bibliography.bib [hunk 6]
```diff
@@ -25,6 +25,25 @@ @article{Bercea2016
   year          = {2016}
 }
 
+@manual{FiredrakeUserManual,
+  author        = {David A. Ham and Paul H. J. Kelly and Lawrence
+Mitchell and Colin J. Cotter and Rob C. Kirby and Koki Sagiyama and
```
Comment (wence-) [inline docs/source/_static/bibliography.bib:30]: I believe @rckirby's publication name is "Robert C. Kirby" FWIW.

Comment (dham) [inline docs/source/_static/bibliography.bib:30]:
```suggestion
Mitchell and Colin J. Cotter and Robert C. Kirby and Koki Sagiyama and
```

File: docs/source/team2.py [hunk 7]
```diff
@@ -103,3 +105,16 @@ def cache_web_image(name, url):
...
     ))
+
+# Create citations file for the Github repository
+citation_rst = env.get_template("citation.rst_t")
+institution_list = [inst[0] for inst in team["dev-institution"].items()]
+institution = ' and '.join(institution_list)
+author_list = list(team['active-team'].keys())
+author_list += list(team['inactive-team'].keys())
+author = ' and '.join(author_list)
+with open("CITATION.rst", "w") as fh:
```
[the remark this answers is not in the document]
Comment (dham) [reply docs/source/team2.py:116]: I think we should just get this merged now.

**Review by dham — APPROVED** [review]

**Review by wence- — APPROVED** [review]

**Pull request merged**